### PR TITLE
Milestone 3: Add and Remove Tasks

### PR DESCRIPTION
**Pull request by CRClothier**
In this checkpoint:
- added the ability to add tasks.
- added the ability to delete tasks.
- removed the placeholder tasks from the tasks array.
- added the ability to change the task names.
- all changes update local storage.

### CODE REVIEW

File: src/deleteTask.js [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+    button.addEventListener('click', () => {
+      items.splice(index, 1);
+      items.forEach((item, i) => {
+        item.index = (i + 1);
+      });
+      localStorage.localData = JSON.stringify(items);
+      createList(items);
+    });
+  });
+}
```
Comment (EricMbouwe) [inline src/deleteTask.js:16]: - [x] The button to clear all completed tasks is not working, let's add a function for the "Clear all completed" button using the `Array.filter()` method.

**Review by EricMbouwe — CHANGES_REQUESTED** [review]
### Hi @CRClothier,

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Highlight!
- Good PR ✔️
- Good commits ✔️
- Code structure ✔️

# Required Changes ♻️ ⛔️

_Check the comments under the review._

## Optional suggestions

_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

You can also consider:

- N/A

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

---

_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: src/deleteTask.js (hunk 1 above)
Comment (henatan99) [reply src/deleteTask.js:16]: Hi @CRClothier , you have made nice progress making the `clear all completed` button work. But it lacks one required behavior. When tasks are deleted the indexes of the remaining tasks should be re-adjusted to be a consecutive integer. 

Screenshot from 2021-12-11 16-07-14

**Review by henatan99 — CHANGES_REQUESTED** [review]
Hi @CRClothier , 

Great work on making the changes requested by a previous reviewer 👏 
You've done well implementing some of the requested changes, but there are still some which aren't addressed yet.


 ## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by EricMbouwe — APPROVED** [review]
### Hi @CRClothier, 

# Status: Approved ✔️
Excellent job on this project. Keep it up 👏
Your project is complete! There is nothing else to say other than... it's time to merge it 💪
Congratulations! 🎉

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**